libvdrtools/indy-wallet/src/storage/mysql/mod.rs

```diff
@@ -811,12 +811,87 @@ impl WalletStorageType for MySqlStorageType {
         credentials: Option<&str>,
         metadata: &[u8],
     ) -> IndyResult<()> {
-        let mut tx = self
-            ._connect(false, config, credentials)
-            .await?
-            .begin()
+        let mut config = config
+            .map(serde_json::from_str::<Config>)
+            .transpose()
+            .to_indy(IndyErrorKind::InvalidStructure, "Malformed config json")?
+            .ok_or(err_msg(
+                IndyErrorKind::InvalidStructure,
+                "Absent config json",
+            ))?;
+
+        let credentials = credentials
+            .map(serde_json::from_str::<Credentials>)
+            .transpose()
+            .to_indy(
+                IndyErrorKind::InvalidStructure,
+                "Malformed credentials json",
+            )?
+            .ok_or(err_msg(
+                IndyErrorKind::InvalidStructure,
+                "Absent credentials json",
+            ))?;
+
+        let mut my_sql_connect_options = MySqlConnectOptions::new()
+            .host(&config.write_host)
+            .username(&credentials.user)
+            .password(&credentials.pass);
+        my_sql_connect_options.log_statements(LevelFilter::Debug);
+
+        let mut pool = MySqlPoolOptions::default()
+            .max_connections(1)
+            .test_before_acquire(false)
+            .connect_with(my_sql_connect_options)
             .await?;
 
+        let mut con = pool.acquire().await?;
+
+        // Basic SQL injection prevention
+        // since we cannot bind the database identifier
+        config.db_name = config.db_name.replace('`', "``");
+
+        sqlx::query(&format!(
+            "CREATE DATABASE IF NOT EXISTS `{}`;",
+            config.db_name
+        ))
+        .execute(&mut con)
+        .await?;
+
+        // Replace the previous single use pool
+        // with the actual one, get a connection
+        // and create the required tables
+        pool = self._connect(false, config, credentials).await?;
+        con = pool.acquire().await?;
+
+        sqlx::query(
+            r#"
+            CREATE TABLE IF NOT EXISTS `items` (
```

Comment:
The `id` is mandatory, yeah. I don't really know how it worked without it there, but it complained until I added it. It is defined for the SQLite storage, which is what made me think it should exist. Having it autoincrement means it must be the `PRIMARY KEY` too. But I'll add a `UNIQUE` constraint on the previous `PRIMARY KEY`

Reply:
On a second thought, the `id` is mandatory because the query for getting all records has an `ORDER BY id` clause. It's not used apart from that. Since this was not being used in any of the code so far, the problem never surfaced.

The ordering is really not important, so instead of doing the unique key shenanigan I modified the `SELECT` to not order the records anymore "¯\_(ツ)_/¯ ".

Reply:
I see, good catch, I didn't know this. So then just putting in attention that our old wallet does not have `id`, field, but that would get created in migrated wallet. I think it should work fine though.

Reply:
This kinda brought me to notice and suggest following, to avoid confusion between the database schema "id" and `id` used in WalletRecord/StorageRecord
https://github.com/hyperledger/aries-vcx/pull/867#discussion_r1248142713

Reply:
I removed the `id` column. It was only used in an irrelevant order by clause.